### PR TITLE
Grad file upload

### PR DESCRIPTION
**Pull request by SodhiA1**

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/graddatacollection/api/batch/exception/FileError.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ */
+public enum FileError {
+  /**
+   * Upload file did not contain any content.
+   */
+  EMPTY_FILE("The uploaded file is empty."),
+  /**
+   * The Invalid mincode header.
+   */
+  MISSING_HEADER("The file header appears to be missing"),
```
Comment (arcshiftsolutions) [inline api/src/main/java/ca/bc/gov/educ/graddatacollection/api/batch/exception/FileError.java:18]: We can drop this

File: api/src/main/java/ca/bc/gov/educ/graddatacollection/api/batch/exception/FileError.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+  /**
+   * The Invalid transaction code header.
+   */
+  INVALID_TRANSACTION_CODE_HEADER("Invalid transaction code on Header record. It must be FFI"),
```
Comment (arcshiftsolutions) [inline api/src/main/java/ca/bc/gov/educ/graddatacollection/api/batch/exception/FileError.java:23]: We can drop this.

File: api/src/main/java/ca/bc/gov/educ/graddatacollection/api/batch/service/GradCourseFileService.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        final IncomingFilesetEntity entity = mapper.toIncomingCRSBatchEntity(fileUpload, schoolID); // batch file can be processed further and persisted.
+        for (final var student : batchFile.getCourseData()) { // set the object so that PK/FK relationship will be auto established by hibernate.
+            final var crsStudentEntity = mapper.toCRSStudentEntity(student, entity);
+            entity.getCourseStudentEntities().add(crsStudentEntity);
+        }
+
+        return craftStudentSetAndMarkInitialLoadComplete(entity, schoolID);
+    }
+
+    @Retryable(maxAttempts = 10, backoff = @Backoff(multiplier = 2, delay = 2000))
```
Comment (arcshiftsolutions) [inline api/src/main/java/ca/bc/gov/educ/graddatacollection/api/batch/service/GradCourseFileService.java:84]: Is this required?

Comment (SodhiA1) [reply api/src/main/java/ca/bc/gov/educ/graddatacollection/api/batch/service/GradCourseFileService.java:84]: No, I'll remove it

**Review by arcshiftsolutions — APPROVED** [review]

**Pull request merged**